stl/inc/chrono

```diff
@@ -619,6 +654,1103 @@ namespace chrono {
+        chrono::year _Year;
+        chrono::month _Month;
+        chrono::weekday_indexed _Weekday_index;
+
+        _NODISCARD static constexpr year_month_weekday _Ymwd_from_days(days _Dp) noexcept {
+            const chrono::year_month_day _Ymd = sys_days{_Dp};
+            const chrono::weekday _Wd         = sys_days{_Dp};
+            const auto _Idx                   = ((static_cast<unsigned>(_Ymd.day()) - 1) / 7) + 1;
+            return {_Ymd.year(), _Ymd.month(), _Wd[_Idx]};
+        }
+
+    public:
+        year_month_weekday() = default;
+        constexpr year_month_weekday(
+            const chrono::year& _Y, const chrono::month& _M, const chrono::weekday_indexed& _Wdi) noexcept
+            : _Year{_Y}, _Month{_M}, _Weekday_index{_Wdi} {}
+        constexpr year_month_weekday(const sys_days& _Sys_days) noexcept
+            : year_month_weekday{_Ymwd_from_days(_Sys_days.time_since_epoch())} {}
+        constexpr explicit year_month_weekday(const local_days& _Local_days) noexcept
+            : year_month_weekday{_Ymwd_from_days(_Local_days.time_since_epoch())} {}
+
+        constexpr year_month_weekday& operator+=(const months& _Months) noexcept;
+        constexpr year_month_weekday& operator-=(const months& _Months) noexcept;
+        constexpr year_month_weekday& operator+=(const years& _Years) noexcept;
+        constexpr year_month_weekday& operator-=(const years& _Years) noexcept;
+
+        _NODISCARD constexpr chrono::year year() const noexcept {
+            return _Year;
+        }
+        _NODISCARD constexpr chrono::month month() const noexcept {
+            return _Month;
+        }
+        _NODISCARD constexpr chrono::weekday weekday() const noexcept {
+            return _Weekday_index.weekday();
+        }
+        _NODISCARD constexpr unsigned index() const noexcept {
+            return _Weekday_index.index();
+        }
+        _NODISCARD constexpr chrono::weekday_indexed weekday_indexed() const noexcept {
+            return _Weekday_index;
+        }
+
+        _NODISCARD constexpr operator sys_days() const noexcept {
+            const sys_days _First = year_month_day{_Year, _Month, day{1}};
+            const days _Diff      = weekday() - chrono::weekday{_First};
+            const days _Days      = _Diff + days{(static_cast<int>(index()) - 1) * 7};
+            return _First + _Days;
+        }
+        _NODISCARD constexpr explicit operator local_days() const noexcept {
+            return chrono::local_days{static_cast<sys_days>(*this).time_since_epoch()};
+        }
+        _NODISCARD constexpr bool ok() const noexcept {
+            if (!_Year.ok() || !_Month.ok() || !_Weekday_index.ok())
+                return false;
+            if (_Weekday_index.index() <= 4)
+                return true;
+            const sys_days _First = year_month_day{_Year, _Month, day{1}};
+            const days _Diff      = weekday() - chrono::weekday{_First};
+            const days _Last      = _Diff + days{29};

```

Comment:
`Diff` is going to be a value between `[0, 6]` so to get what day the first weekday lies on it would be
```cpp
const days _Diff      = weekday() - chrono::weekday{_First} + 1;
```
And then add 4 weeks (28 days). I just rolled the + 1 into the next line.

Reply:
Maybe better variable names and a comment would help?
```cpp
// index() == 5 is only *sometimes* valid
const sys_days _First_of_month    = year_month_day{_Year, _Month, day{1}};
const days _First_of_weekday      = weekday() - chrono::weekday{_First};
const days _Last                  = _Diff + days{29};
```

Reply:
Thanks for the explanation!  That makes much more sense now :)  When originally looking at the algorithm, I wasn't tracing the fact that we know `index == 5` if we get to this point.  I do think some comments/better variable names could be helpful here!